templates/types/streaming/nextjs/app/components/ui/chat/index.ts

```diff
@@ -57,8 +57,9 @@ export type EventData = {
 };
 
 export type AgentEventData = {
-  agent: string;
-  text: string;
+  name: string;
+  msg: string;
```

Comment:
if we change the attributes, we also need to update the TS backend, right?

Reply:
Great catch! We should keep this attributes as is to keep the compatible with TS backend.

Reply:
Here is now the typing:

```typescript
export type ProgressData = {
  id: string;
  total: number;
  current: number;
};

export type AgentEventData = {
  agent: string;
  text: string;
  type: "text" | "progress";
  data?: ProgressData;
};

```

I moved the `ProgressData` to index.ts , remove the snake case pattern and reuse the old attribute names to keep compatible with TS backend. It's minimal and enough for this PR and we can enhance it when implementing for TS